### ISSUE TITLE
A11Y: [calcite-modal] Focus trapping issue with disable-close-button

### ISSUE THREAD

**Issue by BlakeStearman** [Esri/calcite-components#3064]
**Issue**
When `disable-close-button` is set on the `calcite-modal`, focus trapping doesn't work and the user is allowed to tab out of the modal to other page or browser elements.

I've attached a simple test case illustrating this with `disable-close-button` active. If you simply remove the `disable-close-button` attribute, focus trapping will work.

Discovered in beta 65.

**Sample**
https://codepen.io/rpanichakit/pen/RwgQjzW

**Attachments**
[calcite-modal-focus-trap-issue.zip](https://github.com/Esri/calcite-components/files/7187515/calcite-modal-focus-trap-issue.zip)

**Comment by caripizza** [comment]
@BlakeStearman modal also has a `disable-outside-close` attribute, would that work for you here?

**Comment by BlakeStearman** [comment]
Keep open

**Comment by BlakeStearman** [comment]
Keep open for validation&hellip;

**Comment by BlakeStearman** [comment]
Bumping this to keep open. Would be nice to fix for proper A11Y support.

cc @benelan @paulcpederson

**Comment by benelan** [comment]
still reproducible in `beta.77` using the included sample

**Comment by nwhittaker** [comment]
Noticing this issue in the field-apps-web project, as well.

**Comment by geospatialem** [comment]
verified on `next`  ✅
